subtests/docker_cli/run_volumes_selinux/run_volumes_selinux.py

```diff
@@ -72,27 +75,33 @@ def init_container(self, volume=None, volumes_from=None):
         self.sub_stuff['fds'].append(write_fd)
         self.sub_stuff['fds'].append(read_fd)
         dkrcmd = dockercmd.AsyncDockerCmd(self, 'run', subargs)
-        # TODO: Fix use of dkrcmd.stdin when appropriate class mech. available
         dkrcmd.execute(read_fd)
         dkrcmd.stdin = write_fd
         os.close(read_fd)  # no longer needed
-        os.write(write_fd, 'echo "Started"\n')
-        self.failif(not wait_for_output(lambda: dkrcmd.stdout,
-                                        'Started',
-                                        timeout=20),
-                    "Error starting container %s: %s" % (name, dkrcmd))
+
+        # This could take a while if image needs pulling
+        find_name = lambda: str(dc.list_containers_with_name(name)) != []
+        self.failif_ne(utils.wait_for(find_name,
+                                      timeout=self.config['docker_timeout']),
+                       True,  # None == timeout
+                       "Container %s not in 'docker ps' output within"
+                       " timeout: %s" % (name, dc.list_containers()))
+
+        # Don't match echo of the 'echo' command itself
+        self.logdebug('Confirming %s is responding', name)
+        os.write(write_fd, "echo -e '%s'\n" % self.OCTAL_READY)
```

Comment:
Awww, didn't notice my nice commit message answering that?

```
Finally, the shell being
run under docker, has command-echo turned on.  Therefor it's possible
to get a false match, searching output for 'READY' after sending an
'echo READY' as input.  (i.e. it could match feedback of the echo command
itself instead of it's output.)
```